echopype/convert/api.py

```diff
@@ -0,0 +1,553 @@
+import os
+import warnings
+from datetime import datetime as dt
+from pathlib import Path
+
+import fsspec
+from fsspec.implementations.local import LocalFileSystem
+import zarr
+
+from ..utils import io
+
+from ..echodata.echodata import EchoData, XARRAY_ENGINE_MAP
+
+from .parse_azfp import ParseAZFP
+from .parse_ek60 import ParseEK60
+from .parse_ek80 import ParseEK80
+from .set_groups_azfp import SetGroupsAZFP
+from .set_groups_ek60 import SetGroupsEK60
+from .set_groups_ek80 import SetGroupsEK80
+
+MODELS = {
+    "AZFP": {
+        "ext": ".01A",
+        "xml": True,
+        "parser": ParseAZFP,
+        "set_groups": SetGroupsAZFP,
+    },
+    "EK60": {"ext": ".raw", "xml": False, "parser": ParseEK60, "set_groups": SetGroupsEK60},
+    "EK80": {"ext": ".raw", "xml": False, "parser": ParseEK80, "set_groups": SetGroupsEK80},
+    "EA640": {"ext": ".raw", "xml": False, "parser": ParseEK80, "set_groups": SetGroupsEK80},
+}
+
+COMPRESSION_SETTINGS = {
+    'netcdf4': {'zlib': True, 'complevel': 4},
+    'zarr': {'compressor': zarr.Blosc(cname='zstd', clevel=3, shuffle=2)},
+}
+
+DEFAULT_CHUNK_SIZE = {'range_bin': 25000, 'ping_time': 2500}
+
+NMEA_SENTENCE_DEFAULT = ["GGA", "GLL", "RMC"]
+
+
+def _normalize_path(out_f, convert_type, output_storage_options):
+    if convert_type == "zarr":
+        return fsspec.get_mapper(out_f, **output_storage_options)
+    elif convert_type == "netcdf4":
+        return out_f
+
+
+def _validate_path(
+    source_file, file_format, output_storage_options={}, save_path=None
+):
```

Comment:
Following my comment below for `to_file`, I think we can make `_validate_path` to have an option to spit out either one file or multiple files: For use with `open_raw` and `open_mfraw` the output will always be 1 file, but the input may be more than one `source_file`.

Reply:
Agreed. I think we can make a single validate function and do the validation concurrently for multiple files.